x-pack/plugins/security_solution/public/common/hooks/timeline/use_sync_timeline_url_param.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo } from 'react';
+
+import { useUpdateUrlParam } from '../../utils/global_query_string';
+import type { TimelineUrl } from '../../../timelines/store/timeline/model';
+import { timelineSelectors } from '../../../timelines/store/timeline';
+import { TimelineId } from '../../../../common/types';
+import { useShallowEqualSelector } from '../use_selector';
+import { URL_PARAM_KEY } from '../use_url_state';
+
+export const useSyncTimelineUrlParam = () => {
+  const updateUrlParam = useUpdateUrlParam<TimelineUrl>(URL_PARAM_KEY.timeline);
+  const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+  const flyoutTimeline = useShallowEqualSelector((state) => getTimeline(state, TimelineId.active));
+
+  useEffect(() => {
+    updateUrlParam(
+      flyoutTimeline && flyoutTimeline.savedObjectId != null
+        ? {
+            id: flyoutTimeline.savedObjectId != null ? flyoutTimeline.savedObjectId : '',
+            isOpen: flyoutTimeline.show,
+            activeTab: flyoutTimeline.activeTab,
+            graphEventId: flyoutTimeline.graphEventId ?? '',
+          }
+        : null
+    );
+  }, [flyoutTimeline, updateUrlParam]);
```

Comment:
NIT: would it make sense to use the savedObjectId string directly in the dependencies? so we update the url param only when the id really changes.
```suggestion
  }, [flyoutTimeline?.savedObjectId, updateUrlParam]);
```

Reply:
Now I realize there are more parts of the timeline that need to be updated in the url params, but the timeline has a lot of properties that may change, we only need to check 4 of them in the url params. Probably we could spread the ones we need from the selector:
`const { savedObjectId, show, activeTab, graphEventId } = useShallowEqualSelector((state) => getTimeline(state, TimelineId.active));`.
and add them as dependencies in the `useEffect`, they are all primitive types.
It is just a small improvement.